libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts

```diff
@@ -17,23 +19,67 @@ import {
 } from "../../constants/uiConfig.js";
 import { BUCKETS } from "../../constants/bucket.js";
 
+import {
+    hasNoColumns,
+    hasNoMoreThan20ItemsOnMeasuresOrRows,
+    canTableMeasuresOrAttributesAddMoreItems,
+} from "../bucketRules.js";
 import { setBucketTitles } from "../bucketHelper.js";
 import { getTranslation } from "../translations.js";
 import { messages } from "../../../locales.js";
 
+const INCREASE_MEASURE_ITEMS_LIMIT = 100;
+const INCREASE_ROW_ITEMS_LIMIT = 50;
+
 // If you need to edit these icons
 // reflect changes also in gdc-analytical-designer
 // https://github.com/gooddata/gdc-analytical-designer/blob/develop/app/components/buckets/BucketIcon.tsx
 const tableMeasuresIcon = "local:table/bucket-title-measures.svg";
 const tableRowsIcon = "local:table/bucket-title-rows.svg";
 const tableColumnsIcon = "local:table/bucket-title-columns.svg";
 
+function setPivotTableBucketWarningMessages(referencePoint: IExtendedReferencePoint, intl?: IntlShape) {
+    const buckets = referencePoint?.buckets;
+    const updatedUiConfig = cloneDeep(referencePoint?.uiConfig);
+
+    forEach(buckets, (bucket) => {
+        const localIdentifier = bucket?.localIdentifier ?? "";
+        const bucketUiConfig = updatedUiConfig?.buckets?.[localIdentifier];
+
+        if (!bucketUiConfig?.canAddItems) {
+            let warningMessageId;
+            let warningMessageValues = {};
+
+            if (bucket.localIdentifier === BucketNames.COLUMNS) {
+                warningMessageId = messages.columns.id;
+            } else {
+                warningMessageId = messages.measuresAttributes.id;
+
+                if (bucket.localIdentifier === BucketNames.MEASURES) {
+                    warningMessageValues = { limit: "80" };
```

Comment:
evaluate it as newLimit - oldLimit? To keep it auto updated when one of limits will change

Reply:
Done.